packages/framework/src/Model/Product/ProductRepository.php

```diff
@@ -162,10 +154,12 @@ public function addTranslation(QueryBuilder $queryBuilder, $locale)
      * @param \Doctrine\ORM\QueryBuilder $queryBuilder
      * @param int $domainId
      */
-    public function addDomain(QueryBuilder $queryBuilder, $domainId)
+    private function addDomain(QueryBuilder $queryBuilder, $domainId)
     {
-        $queryBuilder->join(ProductDomain::class, 'pd', Join::WITH, 'pd.product = p AND pd.domainId = :domainId');
-        $queryBuilder->setParameter('domainId', $domainId);
+        $queryBuilder->addSelect('pd')
+            ->join('p.domains', 'pd', Join::WITH, 'pd.domainId = :domain');
```

Comment:
Why did you rename the parameter `domainId` to `domain`?

Reply:
ups, fixed